### ISSUE TITLE
Fix Ontag Automation runs

### ISSUE THREAD

**Issue by sowmyav27** [rancher/qa-tasks#429]
### Additions needed
```
Note: 
- The following are not applicable for 2.5.x except for imported RKE2/K3s clusters
- Rancher provisioned RKE2 is applicable from 2.6.5+
- Rancher provisioned k3s is applicable from 2.7.0+
```

- [x] RKE2 custom cluster
- [x] RKE2 Node cluster
- [x] RKE2 EC2 cloud provider enabled node driver
- [x] RKE2 Azure cloud provider enabled node driver
- [x] Hosted Clusters (new)
- [ ] Imported RKE2 clusters
- [x] K3s custom cluster
- [x] K3s Node cluster
- [x] K3s EC2 cloud provider enabled node driver
- [x] K3s Azure cloud provider enabled node driver
- [x] Imported K3s clusters

---
### Fixes needed
- Existing Hosted Clusters - 
- [x] GKE fails to provision --> disable this from OnTag Jenkins
- [x] AKS fails to SSH into node --> update to use v2 and v1. This resolved the AKS issues.
- [x] DO --> Can be removed from Jenkins OnTag run since we don't have access to DO provider.
- [x] Add in new RKE/k3s versions for import clusters --> These versions are hard coded. Could we change it to these as the default values in the jenkins (is there a way we can add the check between 2.5 vs 2.6?)
- - For 2.5
```
k3s - v1.20.15+k3s1
RKE - v1.2.22 
```
- - For 2.6
```
k3s - v1.24.2+k3s1
RKE - v1.3.13 
```

**Comment by slickwarren** [comment]
I have modified the following and have a PR for the ontag manual job:
* removed aks

However on further inspection, I have the following comments:
* GKEv2 ticket is filed for the gke section of ontag. Since that is a separate ticket I think that should be taken up there. In the meantime, I have disabled GKE from ontag since it is failing
* DO is not currently ran as part of the manual ontag job. Do you want me to remove it (and AKS for that matter) from the jenkins ontag job list as well?
* I’m not sure what the last comment means `Add in new RKE/k3s versions for import clusters` as it appears that the ontag job automatically selects the latest version available. Could you clarify what this means?

**Comment by slickwarren** [comment]
Current plan is to fix everything non-rke2 related, have a design meeting about rke2/GO, and continue from there.

**Comment by slickwarren** [comment]
this ticket is necessary in order to run py jobs with go jobs: https://github.com/rancher/qa-tasks/issues/482

**Comment by sowmyav27** [comment]
Latest update after an offline discussion with the team - We need a decision on / work around reusable parameters for GO from the ToU team.  Because currently, there isn’t a good way to run a GO job without a user manually entering the sensitive information. Waiting on info/input from @bmdepesa @igomez06

**Comment by markusewalker** [comment]
Update: Currently, I am exploring adding existing Go jobs into our OnTag automation. What can be done is appending the Go freeform jobs into the OnTag and define different configurations (i.e. RKE2 and K3s node driver and custom cluster). This _should_ achieve the desired functionality.

**Comment by markusewalker** [comment]
The offline discussed implementation details are as follows:

- Utilizing a single YAML file that will be used to provision the added RKE2, K3s node driver and custom cluster configurations
- Updating `Jenkinsfile_ontag_matrix` to account for the new RKE2, K3s node driver and custom cluster configurations

I have a local draft and have shared with @slickwarren. A bit more testing needs to be done before a PR is made, but providing an update.

**Comment by markusewalker** [comment]
2.7 PR: https://github.com/rancher/rancher/pull/40794
2.6 PR: https://github.com/rancher/rancher/pull/41163

**Comment by markusewalker** [comment]
Above PRs merged. Work that is pending:
- AKS certification fixes -> SSH key is not found
- EKS certification fixes -> OnTag passes, but the EKS cluster does not reflect the node or nodegroup within the AWS Management Console. This is because EKS is provisioned independent of a managed nodegroup

More work is needed than initially thought, so this will come later. I'll keep this issue open for tracking that AKS fix.

**Comment by markusewalker** [comment]
2.7 PR to fix AKS/EKS: https://github.com/rancher/rancher/pull/41340
2.6 PR to fix AKS/EKS: https://github.com/rancher/rancher/pull/41341

**Comment by markusewalker** [comment]
Above AKS/EKS PRs merged.

**Comment by markusewalker** [comment]
Above PRs merged. With that, all OnTag fixes/additions defined have been addressed. Closing this issue and if more additions/fixes are needed, we will open a separate issue.
